internal/processor/manifests.go

```diff
@@ -184,24 +184,21 @@ type validateAndStoreManifestOpts struct {
 	ActionBeforeCommit func(*gorp.Transaction) error
 }
 
-func (p *Processor) validateAndStoreManifestCommon(ctx context.Context, account models.ReducedAccount, repo models.Repository, manifest *models.Manifest, manifestBytes []byte, opts validateAndStoreManifestOpts) error {
+func (p *Processor) validateAndStoreManifestCommon(ctx context.Context, account models.ReducedAccount, repo models.Repository, manifest *models.Manifest, digest digest.Digest, manifestBytes []byte, opts validateAndStoreManifestOpts) error {
```

Comment:
It still doesn't feel right to me that this interface does not encode that `digest` is (and must be) computed from `manifestBytes`. I propose adding a new type somewhere like

```
type BytesWithDigest struct {
  bytes []byte
  ownDigest digest.Digest 
}

func NewBytesWithDigest(buf []byte) BytesWithDigest {}
func (b BytesWithDigest) Bytes() []byte {}
func (b BytesWithDigest) Digest() digest.Digest {}
func (b BytesWithDigest) Len() int {}
```
to make the relationship clearer.